sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/TableTests.cs

```diff
@@ -71,6 +73,46 @@ public async Task Table_IfBoundToCustomTableBindingExtension_BindsCorrectly()
             Assert.True(CustomTableBinding<Poco>.DeleteInvoked);
         }
 
+        [RecordedTest]
+        public async Task Table_CreateParameterBindingData_CreatesValidParameterBindingDataObject()
+        {
+            // Arrange
+            var program = new BindToParameterBindingData();
+
+            IHost host = new HostBuilder()
+               .ConfigureDefaultTestHost<BindToParameterBindingData>(program, builder =>
+               {
+                   builder.AddTables();
+               })
+               .Build();
+
+            var jobHost = host.GetJobHost<BindToParameterBindingData>();
+
+            // Act
+            await jobHost.CallAsync(nameof(BindToParameterBindingData.Run));
+            ParameterBindingData result = program.Result;
+
+            Assert.NotNull(result);
+
+            var tableData = result?.Content.ToObjectFromJson<Dictionary<string, object>>();
+
+            // Assert
+            Assert.True(tableData.TryGetValue("TableName", out var tableName));
+            Assert.True(tableData.TryGetValue("Take", out var take));
+            Assert.True(tableData.TryGetValue("Filter", out var filter));
+            Assert.True(tableData.TryGetValue("Connection", out var connection));
+            Assert.True(tableData.TryGetValue("PartitionKey", out var partitionKey));
+            Assert.True(tableData.TryGetValue("RowKey", out var rowKey));
+
+            // Check values
+            Assert.AreEqual("tableName", tableName.ToString());
+            Assert.AreEqual("partitionKey", partitionKey.ToString());
+            Assert.AreEqual("rowKey", rowKey.ToString());
+            Assert.AreEqual("0", take.ToString());
+            Assert.Null(connection);
```

Comment:
Recommend having the test verify Connection and Filter work as expected as well. As it stands, this test wouldn't catch a bug there if the function specified them. If you don't want to do that in this end to end test, you could add another direct unit test for CreateParameterBindingData itself to test various permutations. I.e. a new TablesExtensionConfigProviderTests test class.

Reply:
Added a new test which also verifies connection and filter work as expected